hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/amrmproxy/TestFederationInterceptor.java

```diff
@@ -1242,4 +1255,130 @@ public void testRemoveAppFromRegistryApplicationFailed()
       return null;
     });
   }
+
+  public void testRecoverWithBadSubCluster(final RegistryOperations registryObj) throws IOException, InterruptedException {
+    UserGroupInformation ugi =
+        interceptor.getUGIWithToken(interceptor.getAttemptId());
+
+    ugi.doAs((PrivilegedExceptionAction<Object>) () -> {
```

Comment:
To have such a long function, does it make sense to define the function as a method itself and call it instead a huge nested lambda?

Reply:
Thanks for your help reviewing the code, I will refactor this part of the code.